pycbc/waveform/bank.py

```diff
@@ -294,12 +284,8 @@ def __init__(self, filename, approximant=None, parameters=None,
             for key in data:
                 self.table[key] = data[key]
             # add the compressed waveforms, if they exist
-            if load_compressed and 'compressed_waveforms' in f:
-                self.compressed_waveforms = {}
-                for tmplt_hash in self.table['template_hash']:
-                    self.compressed_waveforms[tmplt_hash] = \
-                        pycbc.waveform.compress.CompressedWaveform.from_hdf(f,
-                            tmplt_hash, load_now=load_compressed_now)
+            if 'compressed_waveforms' in f:
+                self.has_compressed_waveforms = True
```

Comment:
Instead of the if, you can just do `self.has_compressed_waveforms = 'compressed_waveforms' in f` here.